src/api/app/models/bs_request_action_submit.rb

```diff
@@ -119,7 +119,8 @@ def check_action_permission!(skip_source = nil)
     source_package = Package.get_by_project_and_name!(source_project,
                                                       self.source_package,
                                                       opts)
-    return if User.current.can_modify?(source_package)
+    creator_user = User.find_by_login(bs_request.creator)
```

Comment:
please update the minitest suite. It looks like 2 cases are broken, leading to authorization bypass (expected 403, got 200)